src/main/java/org/broadinstitute/hellbender/tools/dragstr/CalibrateDragstrModel.java

```diff
@@ -841,7 +939,7 @@ private static void outputSiteDetails(final PrintWriter writer, final DragstrLoc
     private Stream<org.broadinstitute.hellbender.utils.read.GATKRead> readStream(final ReadsDataSource source, final SimpleInterval interval) {
         final Stream<org.broadinstitute.hellbender.utils.read.GATKRead> unfiltered = interval == null ? Utils.stream(source) : Utils.stream(source.query(interval));
         return unfiltered
-                .filter(read -> (read.getFlags() & DISCARD_FLAG_VALUE) == 0)
+                .filter(read -> (read.getFlags() & DISCARD_FLAG_VALUE) == 0 && read.getAssignedStart() <= read.getEnd())
```

Comment:
Can this filtering step not be applied at the tool level? Also how many empty length reads are typically showing up at this layer/

Reply:
Perhaps.... I did look briefly into it and it seems that they user may have the power to disable filters using the command line if you add it as an tool level filter .... (defined using getDefaultReadFilters() or something like that). what if I want it to be mandatory? Ok I guess that you cannot help it  if you give them the freedom to play with it and  they decide to shoot themselves in the foot, but then we cannot make assumptions in the code.

About the 0 length reads... Certainly not many since we are not encountering issues running the current code with thousands of samples but someone outside the broad bump into this issue causing an exception downstream so it needs to be addressed.

Reply:
I am the person outside Broad who reported this "Start cannot exceed end" error. In all crams I have tried running this tool that used the GATK  indel realignment step,  this error is occurring. The recently sequenced 828 HGDP high coverage crams  all had this error.  The end before start error also occurred in 4789  ADSP cram and the 800 HGDP samples sequenced by NY Genome Center.  Since indel realignment was once in the best practices for the GATK pipeline, this appears to be  a widespread issue for crams run with the indel realignment step.  2000 Korean crams without the GATK indel realignment step ran fine.  I expect the problem reads will have an OC tag which indicates indel realignment occurred.

Reply:
Ah that I guess explains it... in any case the new code should simply ignore those reads. Let's get that bam/cram of your uploaded so that I can add a test for this with empirical data.

Reply:
This is one of the HGDP crams that triggered the error:  ftp:/­/­ftp.­1000genomes.­ebi.­ac.­uk/­vol1/­ftp/­data_collections/­HGDP/­data/­Brahui/­HGDP00001/­alignment/­HGDP00001.­alt_bwamem_GRCh38DH.­20181023.­Brahui.­cram

Reply:
Thanks @jjfarrell